File: 1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var rangeSum = function(nums, n, left, right) {
      const sums = [];
      let mod = Math.pow(10, 9) + 7;

      for (let i = 0; i < nums.length; i++) {
        let cur = nums[i];
        sums.push(cur);
        for (let j = i + 1; j < nums.length; j++) {
          cur += nums[j];
          sums.push(cur);
        }
      }

      sums.sort((a, b) => a - b);

      let result = 0;
      for (let i = left - 1; i < right; i++) {
        result += sums[i];
      }

      return result % mod;
};

File: 1508-range-sum-of-sorted-subarray-sums/1508-range-sum-of-sorted-subarray-sums.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        sums = []
        for i in range(len(nums)):
            cur = nums[i]
            sums.append(cur)
            for j in range(i+1, len(nums)):
                cur += nums[j]
                sums.append(cur)
                
        sums.sort()
        
        return sum(sums[left-1:right]) % (10 ** 9 + 7)